tempodb/backend/gcs/gcs.go

```diff
@@ -30,18 +35,48 @@ type readerWriter struct {
 func New(cfg *Config) (backend.Reader, backend.Writer, backend.Compactor, error) {
 	ctx := context.Background()
 
-	option, err := instrumentation(ctx, storage.ScopeReadWrite)
+	customTransport := http.DefaultTransport.(*http.Transport).Clone()
+
+	transportOptions := []option.ClientOption{}
+	if cfg.Insecure {
+		transportOptions = append(transportOptions, option.WithoutAuthentication())
+		customTransport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	}
+
+	transport, err := google_http.NewTransport(ctx, customTransport, transportOptions...)
 	if err != nil {
-		return nil, nil, nil, err
+		return nil, nil, nil, errors.Wrap(err, "creating transport")
+	}
+	transport = newInstrumentedTransport(transport)
+
+	storageClientOptions := []option.ClientOption{
+		option.WithHTTPClient(&http.Client{
+			Transport: transport,
+		}),
+		option.WithScopes(storage.ScopeReadWrite),
+	}
+	if cfg.Endpoint != "" {
+		storageClientOptions = append(storageClientOptions, option.WithEndpoint(cfg.Endpoint))
 	}
 
-	client, err := storage.NewClient(ctx, option)
+	client, err := storage.NewClient(ctx, storageClientOptions...)
 	if err != nil {
-		return nil, nil, nil, err
+		return nil, nil, nil, errors.Wrap(err, "creating storage client")
 	}
 
 	bucket := client.Bucket(cfg.BucketName)
 
+	// Create bucket if needed.
+	if _, err = bucket.Attrs(ctx); err != nil {
```

Comment:
I'm wondering if this will fail in some installs b/c it requires additional permissions that Tempo is not yet using.

Reply:
Reading the attrs requests `GET /storage/v1/b/<bucket>`, which needs `storage.buckets.get` permission here: https://cloud.google.com/storage/docs/access-control/iam-json   Agree this would be a new permission (doesn't look like it is required to read/write objects).   

This originally started as just an existence check in backend creation in order to fail faster, if there is a problem.  Previously, since the bucket is just a virtual handle, it wouldn't discover a problem until the first block list polling or block written. This change also has the benefit of making the issue a critical startup error, instead of a runtime error.

The bucket creation makes the gcs example easier.  Generally I am in favor of removing all bucket creation logic from tempo, since I think most users will have strong preferences/requirements on the creation of buckets (security/auditing/configuration/etc).   This standpoint applies to S3 as well.

Reply:
As discussed, removed the bucket creation but retained the existence check and fast failing.